nova_vm/src/ecmascript/builtins/ecmascript_function.rs

```diff
@@ -751,15 +835,15 @@ pub(crate) fn make_constructor(
         Function::BoundFunction(_) => unreachable!(),
         // 1. If F is an ECMAScript function object, then
         Function::ECMAScriptFunction(idx) => {
-            // a. Assert: IsConstructor(F) is false.
-            // TODO: How do we separate constructors and non-constructors?
             let data = &mut agent[idx];
+            // a. Assert: IsConstructor(F) is false.
+            assert!(!data.ecmascript_function.constructor_status.is_constructor());
```

Comment:
**nitpick:** This can probably be a debug assert.

Reply:
Done.